projects/core/src/global-message/http-interceptors/handlers/bad-request/bad-request.handler.ts

```diff
@@ -80,7 +81,12 @@ export class BadRequestHandler extends HttpErrorHandler {
     response: HttpErrorResponse
   ): void {
     this.getErrors(response)
-      .filter((e) => e.subjectType === 'cart' && e.reason === 'notFound')
+      .filter(
+        (e) =>
+          e.subjectType === 'cart' &&
+          e.reason === 'notFound' &&
+          !isSelectiveCart(e?.subject)
```

Comment:
1. What do you think about extracting above condition to a separate util function (i.e. `isCartNotFoundError(error)`)? We could reuse it in the http error handler too. This could help associate and track those two places together (see https://github.com/SAP/spartacus/pull/7351/files#diff-962ffd29676c4b32a466258a4324a897R102 ).

2. I guess the condition `!isSelectiveCart(e?.subject)` is not very obvious. I guess it's not a business logic, but a special handling of the backend version that doesn't support a selective cart fetaure. Can we document it in the code (i.e. in the description of the extracted util function)?

3. Bagatelle: Why do really need `?.` ? We don't use it for other properties of `e`.